src/dmd/doc.d

```diff
@@ -2300,6 +2300,27 @@ private void endMarkdownHeading(OutBuffer* buf, size_t iStart, ref size_t iEnd,
     headingLevel = 0;
 }
 
+/****************************************************
+ * End all nested Markdown quotes, if inside any.
+ * Params:
+ *  buf         = an OutBuffer containing the DDoc
+ *  i           = the index within `buf` of the character after the quote text.
+ *                Is incremented by `quoteLevel` when this function ends to \
+ *                account for inserted macro closures.
+ *  quoteLevel  = the current quote level. Is set to `0` when this function ends.
+ *  quoteMacroLevel   = the macro level that the quote was started at. Is set to
+ *                      `0` when this function ends.
+ * Returns: the amount that `i` was moved
+ */
+private size_t endAllMarkdownQuotes(OutBuffer* buf, ref size_t i, ref int quoteLevel, out int quoteMacroLevel)
+{
+    const length = quoteLevel;
+    for (; quoteLevel > 0; --quoteLevel)
+        i = buf.insert(i, ")");
+    quoteMacroLevel = 0;
```

Comment:
What's the point of `quoteMacroLevel`? It's an `out` parameter so it will set its value to the init value, i.e. `0`.

Reply:
`endAllMarkdownQuotes` is called a number of times, so it's a shortcut to allow one line at every call point instead of two:

``` d
endAllMarkdownQuotes(buf, i, quoteLevel);
quoteMacroLevel = 0;
```

Reply:
Hmm, I see. It's not particular clear from the code. It looks like an unused variable inside `endAllMarkdownQuotes`.